### ISSUE TITLE
Analytics auto tracking

### ISSUE THREAD

**Issue by xXBuilderBXx** [isc30/blazor-analytics#8]
Just found out that someone made a javascript for google analytics which helps single page apps like blazor to correctly tracks urls you visit, external links you open and other features here https://github.com/googleanalytics/autotrack it works really well for me now

**Comment by isc30** [comment]
Yes! it works really similarly (by caching the gtag in the global scope and hooking into stuff).
The reason I didn't go with it it's because I wanted to have it open to integrate extra functionality and control from C# (like loading plugins easily, etc).
You can see how I expose gtag [here](https://github.com/isc30/blazor-analytics/blob/master/src/Blazor.Analytics/GoogleAnalytics/Resources/GoogleAnalyticsInterop.ts#L14) and how I get it attached to the navigation system [here](https://github.com/isc30/blazor-analytics/blob/master/src/Blazor.Analytics/GoogleAnalytics/Components/GoogleAnalyticsComponent.cs#L25).
The idea is that you have navigation decoupled from GoogleAnalytics, so you could create a manual tracker and have full control at any point

**Comment by xXBuilderBXx** [comment]
Yea i saw some of the code when i was checking it out but it would be nice if you could add options for the stuff like outbound link tracking. I wanted to get this working for .net core 3.1 but it's a little broken atm #7 but nice work on this.

**Comment by isc30** [comment]
I can't know what's broken in #7 since the DemoApp works well for me. If you could point out that issue I can fix it asap and then we can have a PR for outbound etc

**Comment by isc30** [comment]
ServerSide Blazor is fixed now. We should be able to do this now 😄

**Comment by isc30** [comment]
Hi, are you still interested on this feature?

**Comment by xXBuilderBXx** [comment]
I would not mind it but as long as the main analytics part works it's fine

**Comment by isc30** [comment]
Cool, there is a big change coming to track events, can be implemented easily after it 👍

**Comment by defcon8** [comment]
Yes we need this!!

**Comment by isc30** [comment]
Hi @defcon8 which feature do you need? Is it outbound links?
